Terminal.Gui/Core/Application.cs

```diff
@@ -930,15 +1039,17 @@ static void ResetState ()
 			// Shutdown is the bookend for Init. As such it needs to clean up all resources
 			// Init created. Apps that do any threading will need to code defensively for this.
 			// e.g. see Issue #537
-			// TODO: Some of this state is actually related to Begin/End (not Init/Shutdown) and should be moved to `RunState` (#520)
 			foreach (var t in toplevels) {
 				t.Running = false;
 				t.Dispose ();
 			}
 			toplevels.Clear ();
 			Current = null;
+			Top?.Dispose ();
 			Top = null;
 
+			// BUGBUG: MdiTop is not cleared here, but it should be?
```

Comment:
`MdiTop` is a read only property which check if `Top.IsMdiContainer` and return `Top` or null. So it cannot be set. Maybe the comment can be removed.